cni/network/invoker.go

```diff
@@ -0,0 +1,15 @@
+package network
+
+import (
+	"net"
+
+	"github.com/Azure/azure-container-networking/cni"
+	"github.com/Azure/azure-container-networking/network"
+	cniSkel "github.com/containernetworking/cni/pkg/skel"
+	cniTypesCurr "github.com/containernetworking/cni/pkg/types/current"
+)
+
+type IPAMInvoker interface {
+	Add(args *cniSkel.CmdArgs, nwCfg *cni.NetworkConfig, nwInfo network.NetworkInfo, isDeletePoolOnError bool) (*cniTypesCurr.Result, *cniTypesCurr.Result, error)

```

Comment:
networkinfo is more specific to  azure case.  can we intialize that only as part of azureinvoker and remove from interface calls?

Reply:
can we pass something like options map[string]string?

Reply:
added an options map, but nwInfo seems pretty ubiquitous throughout the ADD/DELETE calls